### ISSUE TITLE
pg num acting exception (was: KeyError)

### ISSUE THREAD

**Issue by jermudgeon** [TheJJ/ceph-balancer#1]
All daemons: ceph version 14.2.16 (5d5ae817209e503a412040d46b3374855b7efe04) nautilus (stable)
```
# ./placementoptimizer.py -v show 
Traceback (most recent call last):
  File "./placementoptimizer.py", line 343, in <module>
    raise Exception(f"on osd.{id} calculated pg num acting: "
Exception: on osd.3 calculated pg num acting: 180 != 179

```

ceph dumps: [ceph-balancer.zip](https://github.com/TheJJ/ceph-balancer/files/5890121/ceph-balancer.zip)

**Comment by jermudgeon** [comment]
Acting != up see https://docs.ceph.com/en/latest/rados/operations/monitoring-osd-pg/ "Note If the Up Set and Acting Set do not match, this may be an indicator that the cluster rebalancing itself or of a potential problem with the cluster." Cluster state is backfilling with 2% degraded, so likely the cause.

**Comment by TheJJ** [comment]
Sorry for the delay.. I've "fixed" it in c8300c531a07fb049ab62b46b0b8d13fcee600b0.
Balancing while being degraded should work with other changes I did in the meantime.

Let me know if it works for you now.
